Examples/VisionExamples/Examples/SafetyAlertsViewController.swift

```diff
@@ -0,0 +1,184 @@
+    }
+}
+
+class SafetyAlertsViewController: UIViewController {
+    private var visionManager: VisionReplayManager!
+    private var visionSafetyManager: VisionSafetyManager!
+
+    private let visionViewController = VisionPresentationViewController()
+
+    private var alertOverspeedingView: UIView!
+
+    private var vehicleState: VehicleState?
+    private var speedLimits: SpeedLimits?
+    private var carCollisions = [CollisionObject]()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Documents directory path with files uploaded via Finder
+        let documentsPath =
+            NSSearchPathForDirectoriesInDomains(.documentDirectory,
+                                                .userDomainMask,
+                                                true).first!
+        let path = documentsPath.appending("/safety-alerts-drawing")
+
+        // create VisionReplayManager with a path to recorded session
+        visionManager = try? VisionReplayManager.create(recordPath: path)
+        // register its delegate
+        visionManager.delegate = self
+
+        // create VisionSafetyManager and register as its delegate to receive safety related events
+        visionSafetyManager = VisionSafetyManager.create(visionManager: visionManager)
+        // register its delegate
+        visionSafetyManager.delegate = self
+
+        // configure Vision view to display sample buffers from video source
+        visionViewController.set(visionManager: visionManager)
+        // add Vision view as a child view
+        addVisionView()
+
+        // add view to draw overspeeding alert
+        addOverspeedingAlertView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        visionManager.start()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        visionManager.stop()
+    }
+
+    deinit {
+        // free up VisionManager's resources
+        visionManager.destroy()
+    }
+
+    private func addVisionView() {
+        addChild(visionViewController)
+        view.addSubview(visionViewController.view)
+        visionViewController.didMove(toParent: self)
+    }
+
+    private func addOverspeedingAlertView() {
+        alertOverspeedingView = UIImageView(image: UIImage(named: "alert"))
+        alertOverspeedingView.isHidden = true
+        alertOverspeedingView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(alertOverspeedingView)
+        NSLayoutConstraint.activate([
+            alertOverspeedingView.topAnchor.constraint(equalToSystemSpacingBelow: view.topAnchor, multiplier: 1),
+            view.trailingAnchor.constraint(equalToSystemSpacingAfter: alertOverspeedingView.trailingAnchor, multiplier: 1)
+        ])
+    }
+
+    // MARK: - Drawing custom collision detection views
+
+    private func clearCollisionStateDrawings() {
```

Comment:
Let's move the body to the `updateCollisionDrawing` method since it's the only place where this code is used.